docroot/modules/custom/sitenow_webform_ais_rfi/src/Plugin/WebformHandler/AisRfiMiddlewareMauiRemotePostWebform.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\sitenow_webform_ais_rfi\Plugin\WebformHandler;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\webform\WebformSubmissionInterface;
+
+/**
+ * Webform handler for AIS RFI Maui.
+ *
+ * @WebformHandler(
+ *   id = "ais_rfi_middleware_maui",
+ *   label = @Translation("AIS RFI Maui"),
+ *   category = @Translation("External"),
+ *   description = @Translation("Posts webform submissions to AIS RFI middleware for Maui."),
+ *   cardinality = \Drupal\webform\Plugin\WebformHandlerInterface::CARDINALITY_UNLIMITED,
+ *   results = \Drupal\webform\Plugin\WebformHandlerInterface::RESULTS_PROCESSED,
+ *   submission = \Drupal\webform\Plugin\WebformHandlerInterface::SUBMISSION_OPTIONAL,
+ *   tokens = TRUE,
+ * )
+ */
+class AisRfiMiddlewareMauiRemotePostWebform extends AisRfiMiddlewareBaseWebformHandler {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return parent::defaultConfiguration() + [
+      'ref_code' => '',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+
+    // Referral Code field.
+    $form['ref_code'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Referral Code'),
+      '#default_value' => $this->configuration['ref_code'],
+      '#description' => $this->t('The middleware refCode. Without this, no data will be sent to the middleware.'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    parent::submitConfigurationForm($form, $form_state);
+
+    // Save the refCode config.
+    $this->configuration['ref_code'] = $form_state->getValue('ref_code');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getRequestData(WebformSubmissionInterface $webform_submission): array {
+    $data = parent::getRequestData($webform_submission);
+
+    // We need refCode to proceed.
+    $ref_code = $this->configuration['ref_code'];
+    if (!$ref_code) {
+      // Log that the refCode is missing.
+      $this->getLogger()->error('AIS RFI Middleware: refCode is missing. No data was sent to the middleware.');
+      return [];
+    }
+    $data['refCode'] = $ref_code;
+
+    return $data;
+  }
```

Comment:
I keep going back and forth on whether we should move the `ref_code` check to happen prior to fetching data with the `parent::getRequestData()`. In this case right now, if we don't have a ref code (or prospector id in the other class) there's no reason to spend time getting the data and no other processes/logging that happens there.

Buuuut, I think I'd still lean toward keeping as it is here and calling the parent first in case the base class ever gets updated with something we wouldn't want to miss. And it's not like it's that time-consuming or costly.

Reply:
Good points. So far there isn't any data that needs to come first, but I've heard mention that the refCode could possibly be altered based on form values or URL parameters in the future.

Ultimately the siteInteractionUuid will always be static so that one could flip.

My guess is that we are still learning what this will actually be, but are being tasked to show it as functional first. Further optimizations like could happen once we know for sure what we are supposed to be passing.

Reply:
There are still a few travis builds before this will get merged. Do you want to flip either or both for now @cory-skeers ?

Reply:
I think given the uncertainty there, let's leave it as is.